tests/test_postgres.py

```diff
@@ -0,0 +1,142 @@
+from itertools import product
+from typing import List
+from typing import Optional
+
+import psycopg2
+import pytest
+from _pytest.mark import ParameterSet
+from pytest_container.container import container_from_pytest_param
+from pytest_container.container import ContainerData
+from pytest_container.container import DerivedContainer
+
+from bci_tester.data import POSTGRES_PASSWORD
+from bci_tester.data import POSTGRESQL_CONTAINERS
+
+
+CONTAINER_IMAGES = POSTGRESQL_CONTAINERS
+
+
+def test_entry_point(auto_container: ContainerData) -> None:
+    # do really nothing here, just check that the container launched
+    assert auto_container.connection.run_expect([0], "ps")
+
+    assert (
+        len(auto_container.inspect.config.entrypoint) == 1
+        and "docker-entrypoint.sh"
+        in auto_container.inspect.config.entrypoint[0]
+    )
+
+
+_other_pg_user = "foo"
+_other_pg_pw = "baz"
+_pgdata = "/opt/pg/data"
+_postgres_user = "postgres"
+
+
+def _generate_test_matrix() -> List[ParameterSet]:
+    params = []
+
+    for pg_cont_param in POSTGRESQL_CONTAINERS:
+        pg_cont = container_from_pytest_param(pg_cont_param)
+        marks = pg_cont_param.marks
+        ports = pg_cont.forwarded_ports
+        params.append(
+            pytest.param(
+                pg_cont, None, POSTGRES_PASSWORD, None, None, marks=marks
+            )
+        )
+
+        for username, pg_user, pw, pgdata in product(
+            (None, _postgres_user),
+            (None, _other_pg_user),
+            (POSTGRES_PASSWORD, _other_pg_pw),
+            (None, _pgdata),
+        ):
+            env = {"POSTGRES_PASSWORD": pw}
+            if pg_user:
+                env["POSTGRES_USER"] = pg_user
+            if pgdata:
+                env["PGDATA"] = pgdata
```

Comment:
why is there no "VOLUME {pgdata}" added to containerfile in this case? what is creating the volume here?